pygmsh/built_in/geometry.py

```diff
@@ -192,6 +192,27 @@ def add_physical_volume(self, volumes, label=None):
         self._add_physical('Volume', volumes, label=label)
         return
 
+    def add_transfinite_lines(self, lines, size):
+        self._GMSH_CODE.append(
+            'Transfinite Line {{{0}}} = {1};'.format(', '.join([l.id for l in lines]), size
+            ))
+
+    def add_transfinite_surface(self, surface, size=None):
+        # a transfinite surface can only have 4 sides
+        assert surface.num_edges == 4
```

Comment:
I suppose this assertion should go _after_ the instance assertion.

Reply:
the `assert` is outside the instance assertion because this can work also with a `SurfaceBase` when `size` is `None`, see previous comments for more details.